src/bartiq/symbolics/sympy_interpreter.py

```diff
@@ -101,49 +101,68 @@ class Round(Function):
 
     @classmethod
     def eval(cls, x, ndigits=None):
-        """Define the function's immediate evaluation in the case where the input is a number."""
-        # Evaluate for explicit number x
-        if x.is_number:
+        # Only evaluate if x is a number
```

Comment:
Why did you touch this class at all? I think it worked just fine and now the logic seems overly complicated, so I'm curious if there was some specific reason for that?